samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/PublishedAssessmentFacadeQueries.java

```diff
@@ -1897,6 +1898,26 @@ public boolean isRandomDrawPart(final Long publishedAssessmentId, final Long sec
 		return false;
 	}
 
+	public boolean isFixedRandomDrawPart(final Long publishedAssessmentId, final Long sectionId) {
+		final String key = SectionDataIfc.AUTHOR_TYPE;
+		final String value = SectionDataIfc.FIXED_AND_RANDOM_DRAW_FROM_QUESTIONPOOL.toString();
+
+		final HibernateCallback<List<PublishedSectionData>> hcb = session -> session
+				.createQuery("select s from PublishedSectionData s, PublishedSectionMetaData m " +
+						" where s = m.section and s.assessment.publishedAssessmentId = :id and s.id = :section and m.label = :key and m.entry = :value")
+				.setParameter("id", publishedAssessmentId)
+				.setParameter("section", sectionId)
+				.setParameter("key", key)
+				.setParameter("value", value)
+				.list();
+		List<PublishedSectionData> l = getHibernateTemplate().execute(hcb);
+
+		if (!l.isEmpty()) {
+			return true;
+		}
+		return false;
```

Comment:
```suggestion
		return (!l.isEmpty());
```
Think this would be the equivalent but better to verify (and not a big deal anyway)

Reply:
Changed!